scripts/verify_privacy_preferences_overhaul_migration.py

```diff
@@ -0,0 +1,782 @@
+        dawns_record.fides_user_device == "dawn219f-20e4-45df-82f7-5eb68a00889f"
+    )  # This is the most recent fides device
+    assert dawns_record.created_at
+    assert dawns_record.updated_at
+    # Privacy preferences were consolidated. Multiple preferences against notice_history and the latest was retained
+    assert dawns_record.preferences == {
+        "preferences": [
+            {"preference": "opt_out", "privacy_notice_history": notice_history_id},
+            {
+                "preference": "opt_out",
+                "privacy_notice_history": functional_notice_history_id,
+            },
+            {
+                "preference": "opt_in",
+                "privacy_notice_history": essential_notice_history_id,
+            },
+        ]
+    }
+
+    print("> Verified Current Privacy Preference V2 migration.")
+
+    existing_served = db.query(DeprecatedLastServedNotice)
+    assert existing_served.count() == 4
+    migrated_current_served = db.query(LastServedNoticeV2).order_by(
+        LastServedNoticeV2.created_at.asc()
+    )
+    assert migrated_current_served.count() == 2
+
+    dawns_served = migrated_current_served.first()
+
+    # Identities combined
+    assert dawns_served.email == "dawn@example.com"
+    assert dawns_served.hashed_email == ProvidedIdentity.hash_value("dawn@example.com")
+    assert dawns_served.phone_number == "+15555555555"
+    assert dawns_served.hashed_phone_number == ProvidedIdentity.hash_value(
+        "+15555555555"
+    )
+    assert dawns_served.fides_user_device == "dawn119f-20e4-45df-82f7-5eb68a00889f"
+    assert dawns_served.hashed_fides_user_device == ProvidedIdentity.hash_value(
+        "dawn119f-20e4-45df-82f7-5eb68a00889f"
+    )
+
+    assert dawns_served.served == {
+        "privacy_notice_history_ids": [notice_history_id, functional_notice_history_id]
+    }
+
+    jane_served = migrated_current_served.offset(1).first()
+    assert jane_served.email is None
+    assert jane_served.hashed_email is None
+    assert jane_served.phone_number is None
+    assert jane_served.hashed_phone_number is None
+    assert jane_served.fides_user_device == "jane119f-20e4-45df-82f7-5eb68a00889f"
+    assert jane_served.hashed_fides_user_device == ProvidedIdentity.hash_value(
+        "jane119f-20e4-45df-82f7-5eb68a00889f"
+    )
+    assert jane_served.created_at
+    assert jane_served.updated_at
+    assert jane_served.served == {
+        "privacy_notice_history_ids": [functional_notice_history_id]
+    }
+
+    print("> Verified LastServedNotice V2 migration.")
+
+
+if __name__ == "__main__":
+    print("> Running Privacy Preferences Migration Script...")
+
+    parser = argparse.ArgumentParser(description="Verify Privacy Preferences Migration")
+    parser.add_argument(
+        "--migrate",
+        dest="reload",
+        action="store_const",
+        const=True,
+        default=False,
+        help="whether or not to redo the migrations and reload the objects",
+    )
+
```

Comment:
This is just a temporary test script illustrating the scenarios considered, will probably remove in the follow up work https://ethyca.atlassian.net/browse/PROD-1529

Reply:
👏 ah how nice, you've already created this followup ticket! 🙏